### PR TITLE
Add description and manufacturer to Device, extract to new file

### PR DESCRIPTION
**Pull request by rly**
Partially handles #258 but we should consider eventually add more structured fields to Device aside from just `manufacturer`.

### CODE REVIEW

**Review by bendichter — APPROVED** [review]

File: core/nwb.file.yaml [hunk 1]
```diff
@@ -258,10 +258,9 @@ groups:
         ADC boards, microscopes, etc.
       quantity: '?'
       groups:
-      - neurodata_type_def: Device
-        neurodata_type_inc: NWBContainer
-        doc: A data acquisition device, e.g. recording system.
-        quantity: '+'
+      - neurodata_type_inc: Device
+        doc: Data acquisition devices.
+        quantity: '*'
```
Comment (rly) [inline core/nwb.file.yaml:263]: Was there a reason why Device had quantity '+' (one or more)? I guess it makes sense that an NWB file should have some data and that data should have some device information attached to it. But then it means our barebones examples of making an NWBFile results in an invalid file (in practice, the validator says such a file is valid). 

I lean toward saying that an NWB file can be valid without any data and without any device.

Comment (bendichter) [reply core/nwb.file.yaml:263]: I agree

Comment (oruebel) [reply core/nwb.file.yaml:263]: I agree

**Comment by bendichter** [comment]
@rly there's a problem here that I did not catch at first. Attributes cannot have the property "quantity". These two properties should instead say `required: False`. This is causing a problem when matnwb tried to auto-generate these classes because it does not read the quantity attribute and instead makes these attributes required, which is the default behavior.

**Comment by rly** [comment]
@bendichter Good catch. Thanks!

**Pull request merged**